dart/lib/vm_service_lib.dart

```diff
@@ -1427,18 +1474,55 @@ class Class extends Obj {
 
   Class._fromJson(Map<String, dynamic> json) : super._fromJson(json) {
     name = json['name'];
-    error = _createObject(json['error']);
+    error = createObject(json['error']);
     isAbstract = json['abstract'];
     isConst = json['const'];
-    library = _createObject(json['library']);
-    location = _createObject(json['location']);
-    superClass = _createObject(json['super']);
-    superType = _createObject(json['superType']);
-    interfaces = new List<InstanceRef>.from(_createObject(json['interfaces']));
-    mixin = _createObject(json['mixin']);
-    fields = new List<FieldRef>.from(_createObject(json['fields']));
-    functions = new List<FuncRef>.from(_createObject(json['functions']));
-    subclasses = new List<ClassRef>.from(_createObject(json['subclasses']));
+    library = createObject(json['library']);
+    location = createObject(json['location']);
+    superClass = createObject(json['super']);
+    superType = createObject(json['superType']);
+    interfaces = new List<InstanceRef>.from(createObject(json['interfaces']));
+    mixin = createObject(json['mixin']);
+    fields = new List<FieldRef>.from(createObject(json['fields']));
+    functions = new List<FuncRef>.from(createObject(json['functions']));
+    subclasses = new List<ClassRef>.from(createObject(json['subclasses']));
+  }
+
+  Map<String, dynamic> toJson() {
+    var json = super.toJson();
+    json["type"] = "Class";
+    json.addAll({
+      'name': name,
+      'abstract': isAbstract,
+      'const': isConst,
+      'library': library?.toJson(),
+      'interfaces': interfaces?.map((f) => f?.toJson())?.toList(),
```

Comment:
I wonder if all these null-aware operators are necessary? I think the vm's impl of the service protocol (and the spec?) sends empty lists, not null. This could then be just `'interfaces': interfaces.map((f) => f.toJson()).toList(),`.

Reply:
How about I update the generator to only use the null aware operators for optional fields? That would also give the desired semantics of not allowing you to serialize objects that are missing required fields (you would get a runtime exception).

Reply:
I went ahead and implemented this